File: intro-langchain-typescript/src/examples/1-fundamentos/2-init-chat-model.ts
import { ChatOpenAI } from "@langchain/openai";
import {
  loadEnvironment,
  getApiKey,
  getEndpoint,
  displayResult,
} from "../../utils/helpers";

loadEnvironment();

async function main(): Promise<void> {
  try {
    const model = new ChatOpenAI({
      model: "gpt-4o",
      temperature: 0,
      maxTokens: 150,
      configuration: {
        baseURL: getEndpoint("GITHUB_MODELS_ENDPOINT"),
        apiKey: getApiKey("GITHUB_MODELS_TOKEN"),
      },
    });

    const response = await model.invoke("Hello, World!");

    displayResult("Init Chat Model (TypeScript)", response.content);
  } catch (error) {
    console.error("Erro:", error instanceof Error ? error.message : error);
  }
}

main();

File: 01-Intro-Langchain/intro-langchain-python/examples/1-fundamentos/2-init-chat-model.py
from langchain.chat_models import init_chat_model
from dotenv import load_dotenv
from pydantic import SecretStr
import os

load_dotenv()

model = init_chat_model(
    model="gpt-4o",
    model_provider="openai",
    base_url=os.environ["GITHUB_MODELS_ENDPOINT"],
    api_key=SecretStr(os.environ["GITHUB_MODELS_TOKEN"])
)

answer_model = model.invoke("Hello World!")
print(answer_model.content)